straxen/plugins/event_processing.py

```diff
@@ -714,36 +742,60 @@ def compute(self, events):
         else:
             s2_top_map_name = "map"
             s2_bottom_map_name = "map"
-
+        
+        regions = [self.ab_region,self.cd_region]
+        
+        if len(self.rel_extraction_eff.names) > 1:
+            se_val = [self.se_gain['gain_ab'],self.se_gain['gain_cd']]
+            ee_cor = [self.rel_extraction_eff['ee_ab'],self.rel_extraction_eff['ee_cd']]
+            avg_se_val = [self.avg_se_gain['gain_avg_ab'],self.avg_se_gain['gain_avg_cd']]
+        else:
+            se_val = [self.rel_light_yield]*2
+            ee_cor = [self.rel_extraction_eff]*2
+            avg_se_val = [self.avg_se_gain]*2
+            
         for peak_type in ["", "alt_"]:
             # S2(x,y) corrections use the observed S2 positions
             s2_positions = np.vstack([events[f'{peak_type}s2_x'], events[f'{peak_type}s2_y']]).T
-
+            result[f"{peak_type}cs2_wo_elifecorr"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_area_fraction_top"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_wo_timecorr"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_bottom"] = np.zeros(len(result['time']))
+            
```

Comment:
Hej sorry already one comment because it catched my eye. Instead of using a dictionary with all this copy past stuff can we move to a structure array via `result = np.zeros(len(events), self.dtype)`. This will increase readability.

Reply:
@Jianyu010 can you make this change?

Reply:
This is still an open issue

Reply:
I guess this slipped now through. Can you still make the change?